### PR TITLE
Fixes #9452: Quality assistant should lock its git local repo access

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/9452

### CODE REVIEW

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
Commit modified

File: scripts/rudder-dev/quality-assistant [hunk 1]
```diff
@@ -35,34 +49,64 @@ except:
...
-    shell("rudder-dev clone "+repo+redirect, "Cloning ncf in a temporary directory")
+    os.chdir(workdir)
+    shell("rudder-dev clone "+repo+redirect, "Cloning "+repo+" in the master directory")
+
+  # check lock
+  lockfile = directory + "/qa.lock"
+  wait = Config.LOCK_TIMEOUT
+  while wait > 0:
+    if os.path.isfile(lockfile):
+      print("Lock present, waiting " + str(wait) + "s")
```
Comment (VinceMacBuche) [inline scripts/rudder-dev/quality-assistant]: You should compute w before and state that you are "waiting for "+ str(w) +"s"

Comment (VinceMacBuche) [inline scripts/rudder-dev/quality-assistant]: (or always wait 5 seconds)

Comment (VinceMacBuche) [inline scripts/rudder-dev/quality-assistant]: ping @peck

**Comment by peckpeck** [comment]
Commit modified

**Pull request merged**